### PR TITLE
[Xamarin.Build.Download] Add a flag to now require secure (HTTPS) URLs

### PR DESCRIPTION
**Pull request by mattleibow**

### CODE REVIEW

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs [hunk 1]
```diff
@@ -47,6 +47,9 @@ public IEnumerable<XamarinBuildDownload> ParseDownloadItems (ITaskItem[] items)
 					//TODO we may be able to determine the kind from the server response
 					continue;
 				}
+				if (!EnsureSecureUrl (item, xbd.Url, allowUnsecureUrls)) {
+					continue;
+				}
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs:52]: This was specifically to follow the same process of a "bad" item - we basically skip it and log an error.

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs [hunk 2]
```diff
@@ -102,6 +105,9 @@ public List<PartialZipDownload> ParsePartialZipDownloadItems (ITaskItem [] items
 					Log.LogCodedError (ErrorCodes.XbdInvalidUrl, "Missing required Url metadata on item {0}", id);
 					continue;
 				}
+				if (!EnsureSecureUrl (part, url, allowUnsecureUrls)) {
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs:108]: Same as above.

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs [hunk 3]
```diff
@@ -145,6 +151,18 @@ public List<PartialZipDownload> ParsePartialZipDownloadItems (ITaskItem [] items
 			return uniqueParts.ToList ();
 		}
 
+		public bool EnsureSecureUrl (ITaskItem item, string url, bool allowUnsecureUrls)
+		{
+			if (!allowUnsecureUrls) {
+				if (url.StartsWith ("http:", StringComparison.OrdinalIgnoreCase)) {
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download/DownloadUtils.cs:157]: This is my "check". Is it sufficient?

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download/ErrorCodes.cs [hunk 4]
```diff
@@ -20,5 +20,7 @@ public static class ErrorCodes
 		public const string XbdInvalidRange = "XBD030";
 		public const string XbdInvalidRangeStart = "XBD031";
 		public const string XbdInvalidRangeEnd = "XBD032";
+
+		public const string XbdUnsecureUrl = "XBD040";
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download/ErrorCodes.cs:24]: A new set of "security" errors.

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
 	<PropertyGroup>
 		<XamarinBuildDownloadDir Condition="'$(XamarinBuildDownloadDir)' != '' and !HasTrailingSlash('$(XamarinBuildDownloadDir)')">$(XamarinBuildDownloadDir)\</XamarinBuildDownloadDir>
+		<XamarinBuildDownloadAllowUnsecure Condition="'$(XamarinBuildDownloadAllowUnsecure)' == ''">false</XamarinBuildDownloadAllowUnsecure>
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download/Xamarin.Build.Download.targets:13]: The default is to be `false`.

File: Util/Xamarin.Build.Download/source/Xamarin.Build.Download.Tests/Test.cs [hunk 6]
```diff
@@ -448,9 +448,90 @@ public void testAndroidAarAdded (bool useAndroidSdk)
...
+			AddCoreTargets (prel);
+
+			var project = new ProjectInstance (prel);
+			var log = new MSBuildTestLogger ();
+
+			var success = BuildProject (engine, project, "XamarinBuildDownloadGetItemsToDownload", log);
+
+			var itemToDownload = project.GetItems ("XamarinBuildDownloadItemToDownload");
+
+			Assert.IsEmpty (itemToDownload);
```
Comment (mattleibow) [inline Util/Xamarin.Build.Download/source/Xamarin.Build.Download.Tests/Test.cs:486]: I am not sure if it is just me, but my build does not fail when there is an error logged.

**Review by Redth — APPROVED** [review]

**Pull request merged**